tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/SilentAuthTests.cs

```diff
@@ -155,6 +155,7 @@ public async Task SilentAuth_TokenCacheRemainsPersistent_Async()
             Assert.IsFalse(at1.Equals(at2, System.StringComparison.InvariantCultureIgnoreCase));
             Assert.IsFalse(at1.Equals(at3, System.StringComparison.InvariantCultureIgnoreCase));
             Assert.IsFalse(at2.Equals(at3, System.StringComparison.InvariantCultureIgnoreCase));
+            Assert.AreEqual(TokenSource.IdentityProvider, authResult.AuthenticationResultMetadata.TokenSource);
         }
```

Comment:
can you assert earlier, like after line 133, that the TokenSource is `TokenSource.Cache`...I guess finding one that does AT silent and pulls from the cache. I would just add this check in all the tests, what about the interactive tests and broker tests?

Reply:
Or after line 145, that AcquireTokenSilent gets a token from teh cache.

Reply:
That one is withForceRefresh(True) so it would not get the token from the cache. Right?

Reply:
Added in the other method where force refresh was false.

Reply:
the assertion at line 158 is correct - force_refresh means "do not look for an AT in the cache".